src/ai/backend/manager/repositories/user_resource_policy/db_source/db_source.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Optional
+
+import sqlalchemy as sa
+
+from ai.backend.common.exception import BackendAIError, UserResourcePolicyNotFound
+from ai.backend.common.metrics.metric import DomainType, LayerType
+from ai.backend.common.resilience.policies.metrics import MetricArgs, MetricPolicy
+from ai.backend.common.resilience.policies.retry import BackoffStrategy, RetryArgs, RetryPolicy
+from ai.backend.common.resilience.resilience import Resilience
+from ai.backend.manager.data.resource.types import UserResourcePolicyData
+from ai.backend.manager.models.resource_policy import UserResourcePolicyRow
+from ai.backend.manager.services.user_resource_policy.actions.modify_user_resource_policy import (
+    UserResourcePolicyModifier,
+)
+from ai.backend.manager.services.user_resource_policy.types import UserResourcePolicyCreator
+
+if TYPE_CHECKING:
+    from ai.backend.manager.models.utils import ExtendedAsyncSAEngine
+
+user_resource_policy_db_source_resilience = Resilience(
+    policies=[
+        MetricPolicy(
+            MetricArgs(domain=DomainType.DB_SOURCE, layer=LayerType.USER_RESOURCE_POLICY_DB_SOURCE)
+        ),
+        RetryPolicy(
+            RetryArgs(
+                max_retries=5,
+                retry_delay=0.1,
+                backoff_strategy=BackoffStrategy.FIXED,
+                non_retryable_exceptions=(BackendAIError,),
+            )
+        ),
+    ]
+)
+
+
+class UserResourcePolicyDBSource:
+    """
+    Database source for user resource policy operations.
+    Handles all database operations for user resource policies.
+    """
```

Comment:
Why is serializable applied here? It seems like applying Read Committed this time would be worthwhile, don't you think?